frontend/src/components/App/AppComponent.jsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+// import { GreetingContainer } from '../Greeting/GreetingContainer';
+import { Route } from 'react-router-dom';
+// import { LoginFormContainer } from '../Session/LoginFormContainer';
+// import { SignupFormContainer } from '../Session/SignupFormContainer';
+import { createNewUser, login, logout } from '../../actions/user_actions';
+import { AuthRoute } from '../util/route_util';
+
+import NavBarContainer from '../NavBar/NavBarContainer';
+import MainBarContainer from '../MainBar/MainBarContainer';
+
+export class AppComponent extends React.Component { 
+  constructor(props) {
+    super(props);
+    this.timeout = null;
+    this.toggleUserDrop = this.toggleUserDrop.bind(this);
```

Comment:
không dùng `bind` ở đây được ko, giải thích tại sao dùng nó? Có cách nào khác ngoài bind ko?

Reply:
do es6 sử dụng strict mode, truyền callback thì mất đi context => khi event xảy ra this sẽ trở về default binding, mang giá trị undefined, chính vì vậy mà chúng ta phải bind component instance vào trong constructor
nếu không dùng thì dùng arrow function thôi a